src/Compilers/CSharp/Portable/Parser/Lexer_StringLiteral.cs

```diff
@@ -535,19 +520,10 @@ private void ScanInterpolatedStringLiteralHoleBalancedText(char endingChar, bool
                         case '$':
                             if (_lexer.TextWindow.PeekChar(1) == '"' || _lexer.TextWindow.PeekChar(1) == '@' && _lexer.TextWindow.PeekChar(2) == '"')
                             {
-                                bool isVerbatimSubstring = _lexer.TextWindow.PeekChar(1) == '@';
-                                var interpolations = (ArrayBuilder<Interpolation>?)null;
-                                var info = default(TokenInfo);
-                                bool wasVerbatim = _isVerbatim;
-                                try
-                                {
-                                    _isVerbatim = isVerbatimSubstring;
-                                    ScanInterpolatedStringLiteralTop(interpolations, ref info, closeQuoteMissing: out _);
```

Comment:
the act of recursing directly into sub functions was adding a lot of complexity in teh raw-string-literal side of things.  It's much simpler for the interpolation-sub-scanner to just recurse directly into the lexer so that all state is fresh and appropriate to each sub-item scanned.

Reply:
note: we have a *lot* of tests for this area (several of my previous PRs beefed this up a ton).

Reply:
So was the old way trying to reuse the same `InterpolatedStringScanner` instance between the parent and nested interpolated strings? I don't see why else it would work the way it did..

Reply:
Correct.  But i posit there is anti-value in doing that, esp. given how recursive the processing is of nested interpolations is.